lib-impl/src/main/java/org/ergoplatform/appkit/impl/OutBoxImpl.scala

```diff
@@ -1,16 +1,27 @@
 package org.ergoplatform.appkit.impl
 
 import org.ergoplatform.ErgoBoxCandidate
-import org.ergoplatform.appkit.{ErgoId, ErgoToken, OutBox}
+import org.ergoplatform.appkit.{ErgoToken, ErgoValue, Iso, OutBox}
 import scorex.util.ModifierId
+import sigmastate.Values
 
-class OutBoxImpl(_ctx: BlockchainContextImpl,
-                 _ergoBoxCandidate: ErgoBoxCandidate) extends OutBox {
+import java.util
+
+class OutBoxImpl(_ctx: BlockchainContextImpl, _ergoBoxCandidate: ErgoBoxCandidate) extends OutBox {
   override def getValue: Long = _ergoBoxCandidate.value
 
   override def getCreationHeight: Int = _ergoBoxCandidate.creationHeight
 
-  override def token(id: ErgoId): ErgoToken = null
+  override def getTokens: util.List[ErgoToken] = Iso.isoTokensListToPairsColl.from(_ergoBoxCandidate.additionalTokens)
+
+  override def getRegisters: util.List[ErgoValue[_]] = {
+    val registers = _ergoBoxCandidate.additionalRegisters.values.toIndexedSeq
+    Iso.JListToIndexedSeq(Iso.isoErgoValueToSValue).from(registers)
```

Comment:
Just a hint, is Scala you can use convertTo method when you know the resulting type. In most of the cases the compiler will summon the necessary iso automatically.

Reply:
Yes, was in java mode while writing some of these methods 😄 
I'll try to make this look a bit cleaner.